File: src/main/java/no/nav/pto/veilarbportefolje/domene/VeilederId.java
package no.nav.pto.veilarbportefolje.domene;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;

@Getter
public class VeilederId {
    public final String veilederId;

    @Override
    public String toString() {
        return veilederId;
    }

    public static VeilederId of(String veilederId) {
        return new VeilederId(veilederId);
    }

    @JsonCreator
    public VeilederId(@JsonProperty("veilederId") String veilederId) {
        this.veilederId = veilederId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        VeilederId that = (VeilederId) o;

        return veilederId.equals(that.veilederId);
    }

    @Override
    public int hashCode() {
        return veilederId.hashCode();
    }
}

File: src/main/java/no/nav/pto/veilarbportefolje/domene/VeilederId.kt
package no.nav.pto.veilarbportefolje.domene

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class VeilederId(
    @JsonProperty("veilederId") val value: String?
) {
    companion object {
        @JvmStatic
        @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
        fun of(veilederId: String?) = VeilederId(veilederId)

        @JvmStatic
        fun veilederIdOrNull(veilederId: String?): VeilederId? =
            veilederId?.let { VeilederId(it) }
    }

    override fun toString() = value.orEmpty()
}
